src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -467,6 +467,7 @@ OCC::Result<void, QString> OCC::CfApiWrapper::registerSyncRoot(const QString &pa
     const auto version = std::wstring(providerVersion.toStdWString().data());
 
     CF_SYNC_REGISTRATION info;
+    info.StructSize = sizeof(info);
```

Comment:
Ah good catch, did that create issues in practice?

One caveat with their API to investigate is that sometimes you need such StructSize members to also encompass the size of all the variable size members so that'd be something along the lines of `sizeof(info) + (name.size() + version.size() + ...) * sizeof(wchar_t)`.

I doubt this is the case here because it's not all in one block of allocation but since you're looking into this I guess something is fishy so better safe than sorry.

Reply:
So this is #3158 .
I was able to reproduce this issue and this fix seems enough to avoid the problem.

I was indeed wondering if I should not try to compute the real size. I will try that.

There is also one more field that we do not initialize yet. I will also need to investigate that.

Reply:
I had missed that other struct also have this like at line 46 .

Reply:
> There is also one more field that we do not initialize yet. I will also need to investigate that.

This on the other hand might not be an issue. Some of the fields are documented as optional. Just need to make sure it is and that it has a known value.

Reply:
Indeed, `ProviderId` from `CF_SYNC_REGISTRATION` has random values and does not cause harm.
I am trying to find the value I should put there.

Reply:
So I will explicitly initialize this last parameter that is not set as optional. Better have some values that uninitialized random garbage.